flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/DefaultScheduler.java

```diff
@@ -258,22 +270,32 @@ protected void scheduleOrUpdateConsumersInternal(final ExecutionVertexID produce
 	// ------------------------------------------------------------------------
 
 	@Override
-	public void allocateSlotsAndDeploy(final Collection<ExecutionVertexDeploymentOption> executionVertexDeploymentOptions) {
+	public void allocateSlotsAndDeploy(final List<ExecutionVertexDeploymentOption> executionVertexDeploymentOptions) {
 		validateDeploymentOptions(executionVertexDeploymentOptions);
 
-		final Map<ExecutionVertexID, ExecutionVertexDeploymentOption> deploymentOptionsByVertex =
-			groupDeploymentOptionsByVertexId(executionVertexDeploymentOptions);
+		final List<ExecutionVertexDeploymentOption> sortedDeploymentOptions =
```

Comment:
What are the benefits to sort before `deployAll`?

Reply:
The "transition ... to SCHEDULED" log can be in order.

Reply:
ok